include/aws/common/byte_buf.h

```diff
@@ -191,20 +201,45 @@ static inline struct aws_byte_buf aws_byte_buf_from_array(const uint8_t *bytes,
     return buf;
 }
 
+static inline struct aws_byte_buf aws_byte_buf_from_empty_array(const void *bytes, size_t capacity) {
```

Comment:
Would it be more useful to have a variant with both length and capacity args?

Reply:
I don't think so, so far all use cases I've seen have followed one of the two patterns. I've yet to see a len that isn't 0 or capacity.